### PR TITLE
Set max number of splits

### PR DESCRIPTION
**Pull request by lhoestq**
To avoid cases with 300 splits = 300 load_dataset() * 300 data files resolutions = 90k calls to the Hub API to list files

### CODE REVIEW

File: services/worker/src/worker/job_runners/config/split_names.py [hunk 1]
```diff
@@ -83,10 +87,17 @@ def compute_split_names_from_streaming_response(
             f"Cannot get the split names for the config '{config}' of the dataset.",
             cause=err,
         ) from err
+    if len(split_name_items) > max_number:
+        split_examples = ", ".join([split_name_item["split"] for split_name_item in split_name_items[:5]])
+        raise DatasetWithTooManySplitsError(
+            f"The {config} config contains {len(split_name_items)} while it should generally contain 3 splits maximum (train/validation/test). "
+            f"If the splits {split_examples}... are not used to differentiate between training and evaluation, please consider defining configs of this dataset instead. "
```
Comment (severo) [inline services/worker/src/worker/job_runners/config/split_names.py:94]: excellent!

File: services/worker/src/worker/config.py [hunk 2]
```diff
@@ -259,6 +259,22 @@ def from_env(cls) -> "ConfigNamesConfig":
             )
 
 
+SPLIT_NAMES_MAX_NUMBER = 30
```
Comment (severo) [inline services/worker/src/worker/config.py:262]: frankly, I would hardcode it

Comment (lhoestq) [reply services/worker/src/worker/config.py:262]: I like that it's at the same place as other constants.

But yea having a config is overkill, we can change that later

**Review by albertvillanova — COMMENTED** [review]
You can merge the main branch once the PR with the vulnerability fix has been merged:
- #2774

**Pull request merged**